source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/form9/ExpropriationForm9.tsx

```diff
@@ -0,0 +1,107 @@
+import { Formik, FormikHelpers, FormikProps } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+
+import { Button } from '@/components/common/buttons';
+import { Input } from '@/components/common/form';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import FormItem from '@/components/common/form/FormItem';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { RestrictContactType } from '@/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent';
+import FilePropertiesTable from '@/components/filePropertiesTable/FilePropertiesTable';
+import { Api_AcquisitionFile } from '@/models/api/AcquisitionFile';
+import { Api_PropertyFile } from '@/models/api/PropertyFile';
+
+import { ExpropriationForm9Model } from '../models';
+import { ExpropriationForm9YupSchema } from './ExpropriationForm9YupSchema';
+
+export interface IExpropriationForm9Props {
+  acquisitionFile: Api_AcquisitionFile;
+  onGenerate: (values: ExpropriationForm9Model) => Promise<void>;
+}
+
+export const ExpropriationForm9: React.FC<IExpropriationForm9Props> = ({
+  acquisitionFile,
+  onGenerate,
+}) => {
+  const onSubmit = async (
+    values: ExpropriationForm9Model,
+    formikHelpers: FormikHelpers<ExpropriationForm9Model>,
+  ) => {
+    try {
+      await onGenerate(values);
+    } finally {
+      formikHelpers?.setSubmitting(false);
+    }
+  };
+
+  const onCancelClick = (formikProps: FormikProps<ExpropriationForm9Model>) => {
+    formikProps.resetForm();
+  };
+
+  const onGenerateClick = (formikProps: FormikProps<ExpropriationForm9Model>) => {
+    formikProps.setSubmitting(true);
+    formikProps.submitForm();
```

Comment:
I wonder if this should reset the form too?

Reply:
By not resetting the form it allows for re-printing the Form after tweaking some values right? So I rather wait for user feedback on this...